### PR TITLE
[docs] Add section about disabling columns panel

### PR DESCRIPTION
**Pull request by MBilalShafi**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

https://mui-org.slack.com/archives/C04U3R2V9UK/p1691095016645389

### CODE REVIEW

File: packages/grid/x-data-grid/src/hooks/features/columns/gridColumnsInterfaces.ts [hunk 1]
```diff
@@ -30,4 +30,4 @@ export type GridColumnsRawState = Omit<GridColumnsState, 'lookup'> & {
 
 export type GridHydrateColumnsValue = GridColumnsRawState;
 
-export type GridColumnVisibilityModel = Record<GridRowId, boolean>;
+export type GridColumnVisibilityModel = Record<GridColDef['field'], boolean>;
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/hooks/features/columns/gridColumnsInterfaces.ts:33]: This seemed incorrect as the model accepts a Record of `field` keys

File: docs/data/data-grid/column-visibility/column-visibility.md [hunk 2]
```diff
@@ -60,6 +60,19 @@ The user can then choose which columns are visible using the _Columns_ button.
 
 {{"demo": "ColumnSelectorGrid.js", "bg": "inline"}}
 
+### Disable the column visibility panel
+
+Sometimes, you may want to simply disable the columns panel or control the visible columns programatically based on the application state like the type of user logged in.
+To disable the column visibility panel, set the prop `disableColumnSelector={true}` and use the [`columnVisibilityModel`](#controlled-visible-columns) prop to control the visible columns.
```
Comment (DanailH) [inline docs/data/data-grid/column-visibility/column-visibility.md:66]:
```suggestion
Sometimes, we may want to disable the columns panel or control the visible columns programmatically based on the application state.
To disable the column visibility panel, set the prop `disableColumnSelector={true}` and use the [`columnVisibilityModel`](#controlled-visible-columns) prop to control the visible columns.
```

File: docs/data/data-grid/column-visibility/column-visibility.md [hunk 3]
```diff
@@ -60,6 +60,19 @@ The user can then choose which columns are visible using the _Columns_ button.
...
+
+```tsx
+<DataGrid disableColumnSelector columnVisibilityModel={columnVisibilityModel} />
+```
+
+In the following demo, columns panel is disabled and access to columns `id`, `quantity`, and `filledQuantity` is only allowed to the user type `Admin`. Try changing the user type to see how the visible columns change.
```
Comment (DanailH) [inline docs/data/data-grid/column-visibility/column-visibility.md:72]:
```suggestion
In the following demo, the columns panel is disabled, and access to columns `id`, `quantity`, and `filledQuantity` is only allowed for the `Admin` type user.
```

**Review by DanailH — APPROVED** [review]

File: docs/data/data-grid/column-visibility/column-visibility.md [hunk 4]
```diff
@@ -60,6 +60,19 @@ The user can then choose which columns are visible using the _Columns_ button.
 
 {{"demo": "ColumnSelectorGrid.js", "bg": "inline"}}
 
+### Disable the column visibility panel
+
+Sometimes, we may want to disable the columns panel or control the visible columns programmatically based on the application state.
```
Comment (MBilalShafi) [inline docs/data/data-grid/column-visibility/column-visibility.md:65]:
```suggestion
Sometimes, the intention is to disable the columns panel or control the visible columns programmatically based on the application state.
```

**Pull request merged**